client/state/posts/likes/reducer.js

```diff
@@ -27,135 +27,87 @@ import {
+				iLike: true,
+				found: state.found + 1,
 			};
 		},
-		[ POST_UNLIKE ]: ( state, { siteId, postId } ) => {
-			const currentLike = get( state, [ siteId, postId ], undefined );
-			if ( ! currentLike ) {
-				return state;
-			}
-
-			if ( ! currentLike.iLike ) {
+		[ POST_UNLIKE ]: state => {
+			if ( ! state.iLike ) {
 				return state;
 			}
 
 			return {
-				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					[ postId ]: {
-						likes: currentLike.likes,
-						iLike: false,
-						found: currentLike.found - 1,
-					},
-				},
+				likes: state.likes,
+				iLike: false,
+				found: Math.max( 0, state.found - 1 ),
 			};
 		},
-		[ POST_LIKES_ADD_LIKER ]: ( state, { siteId, postId, likeCount, liker } ) => {
-			const currentLike = get( state, [ siteId, postId ], {
-				likes: undefined,
-				iLike: false,
-				found: 0,
-			} );
-
-			const hasLiker = some( currentLike.likes, like => like.ID === liker.ID );
+		[ POST_LIKES_ADD_LIKER ]: ( state, { likeCount, liker } ) => {
+			const hasLiker = some( state.likes, like => like.ID === liker.ID );
 
-			if ( currentLike.likeCount === likeCount && hasLiker ) {
+			if ( state.likeCount === likeCount && hasLiker ) {
 				// if the like count matches and we already have this liker, bail
 				return state;
 			}
 
-			let likes = currentLike.likes;
+			let likes = state.likes;
 			if ( ! hasLiker ) {
-				likes = [ liker, ...( currentLike.likes || [] ) ];
+				likes = [ liker, ...( state.likes || [] ) ];
 			}
 
 			return {
-				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					[ postId ]: {
-						likes,
-						iLike: currentLike.iLike,
-						found: likeCount,
-					},
-				},
+				likes,
+				iLike: state.iLike,
+				found: likeCount,
 			};
 		},
-		[ POST_LIKES_REMOVE_LIKER ]: ( state, { siteId, postId, likeCount, liker } ) => {
-			const currentLike = get( state, [ siteId, postId ], {
-				likes: undefined,
-				iLike: false,
-				found: 0,
-			} );
+		[ POST_LIKES_REMOVE_LIKER ]: ( state, { likeCount, liker } ) => {
+			const hasLiker = some( state.likes, like => like.ID === liker.ID );
 
-			const hasLiker = some( currentLike.likes, like => like.ID === liker.ID );
-
-			if ( currentLike.likeCount === likeCount && ! hasLiker ) {
+			if ( state.likeCount === likeCount && ! hasLiker ) {
 				// if the like count matches and we don't have this liker, bail
 				return state;
 			}
 
-			let likes = currentLike.likes;
+			let likes = state.likes;
 			if ( hasLiker ) {
-				likes = dropWhile( currentLike.likes, l => liker.ID === l.ID );
+				likes = dropWhile( state.likes, l => liker.ID === l.ID );
 			}
 
 			return {
-				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					[ postId ]: {
-						likes,
-						iLike: currentLike.iLike,
-						found: likeCount,
-					},
-				},
+				likes,
+				iLike: state.iLike,
+				found: likeCount,
 			};
 		},
 	},
-	itemsSchema
+	itemSchema
 );
 
+const postIdReducer = keyedReducer( 'postId', itemReducer );
```

Comment:
You could probably leave the schema how it was if you applied it like this:
```js
export const items = withSchemaValidation(
  keyedReducer( 'siteId', keyedReducer( 'postId' , itemReducer ) ),
  schema
);
```

Reply:
yeah... played with that, but ... dunno.